src/systemctl/systemctl-set-environment.c

```diff
@@ -8,6 +8,44 @@
 #include "systemctl-util.h"
 #include "systemctl.h"
 
+static int json_transform_message(sd_bus_message *m, JsonVariant **ret) {
+        _cleanup_(json_variant_unrefp) JsonVariant *v = NULL, *w = NULL;
+        char *text, *sep;
+        int r;
+
+        assert(m);
+        assert(ret);
+
+        r = sd_bus_message_enter_container(m, SD_BUS_TYPE_ARRAY, "s");
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        while ((r = sd_bus_message_read_basic(m, SD_BUS_TYPE_STRING, &text)) > 0) {
+                sep = strchr(text, '=');
+                if (!sep)
+                        return log_error_errno(SYNTHETIC_ERRNO(EUCLEAN),
+                                            "Invalid environment block");
+
+                *sep = '\0';
```

Comment:
`*sep++ = ...` and remove the `+ 1` below - also can there be whitespace before/after `=`? If so, it should be pruned?

Reply:
According to the man pages, having whitespace before/after `=` seems to be a syntax error when using `set-environment`. The man pages for bash show the same syntax, so when the environment variable is imported using `import-environment`, I cannot see how it's possible to have whitespace before/after `=`. Please correct me if I'm wrong.

Reply:
it's fine then